test/integration/yurttunnel_test.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	anpserver "sigs.k8s.io/apiserver-network-proxy/pkg/server"
+
+	ta "github.com/alibaba/openyurt/pkg/yurttunnel/agent"
+	hw "github.com/alibaba/openyurt/pkg/yurttunnel/handlerwrapper"
+	tr "github.com/alibaba/openyurt/pkg/yurttunnel/handlerwrapper/tracerequest"
+	ts "github.com/alibaba/openyurt/pkg/yurttunnel/server"
+)
+
+const (
+	ResponseForRegularRequest = "Dummy Server"
+	DummyServerPort           = 9515
+	ServerMasterPort          = 9516
+	ServerMasterInsecurePort  = 9517
+	ServerAgentPort           = 9518
+	HTTPPath                  = "yurttunel-test"
+	RootCAFile                = "ca.pem"
+	GeneircCertFile           = "cert.pem"
+	GenericKeyFile            = "key.pem"
+	InterceptorServerUDSFile  = "interceptor-proxier.sock"
+)
+
+func genCAPool(t *testing.T, CAPath string) *x509.CertPool {
+	caCertPEM, err := ioutil.ReadFile(CAPath)
+	if err != nil {
+		t.Fatalf("fail to load the CA: %v", err)
+	}
+	roots := x509.NewCertPool()
+	ok := roots.AppendCertsFromPEM(caCertPEM)
+	if !ok {
+		t.Fatal("fail to append the ca PEM to cert pool")
+	}
+	return roots
+}
+
+func genCert(t *testing.T, certPath, keyPath string) tls.Certificate {
+	cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		t.Fatalf("fail to load the dummy server cert: %v", err)
+	}
+	return cert
+}
+
+func startDummyServer(t *testing.T) {
+	m := http.NewServeMux()
+	m.HandleFunc("/"+HTTPPath, func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(ResponseForRegularRequest))
+	})
+
+	s := &http.Server{
+		TLSConfig: &tls.Config{
+			Certificates: []tls.Certificate{
+				genCert(t, GeneircCertFile, GenericKeyFile),
+			},
+			ClientCAs: genCAPool(t, RootCAFile),
+		},
+		Addr:    fmt.Sprintf(":%d", DummyServerPort),
+		Handler: m,
+	}
+
+	klog.Info("[TEST] dummy-server is listening on :9515")
```

Comment:
how about use var DummyServerPort  to substitute hard code 9515

Reply:
Yes. Will replace the magic number with the var.